src/pdm/cli/commands/list.py

```diff
@@ -335,17 +322,20 @@ class Listable:
     def __init__(self, dist: im.Distribution, groups: set[str]):
         self.dist = dist
 
-        self.name = _get_metadata_key(dist, "Name")
+        self.name = dist.metadata.get("Name")
```

Comment:
I object to the use of `.get`, which is not exposed in the [public protocol of importlib-metadata](https://github.com/python/importlib_metadata/blob/f3901686abc47853523f3b211873fc2b9e0c5ab5/importlib_metadata/_meta.py#L11) but is only an implementation detail which we shouldn't rely on.